devolving_music/lib/song_scores.py

```diff
@@ -110,42 +137,86 @@ def get_quality_list(self,
+            get_informed=False,
+            informed_threshold=1) -> List["ScoreSuite"]:
         """
-        Returns a list of song suites ordered by low information songs with the lowest information score 
-        are randomly shuffled before being added to the list
-        
+        Returns a list of song suites ordered by low information songs with the lowest information score
+        randomly shuffled before being added to the list
+
         """
-        score_suite_list.sort(key=lambda sub: sub.info_score)
+        score_suite_list = sorted(score_suite_list, key=lambda sub: sub.info_score)
         rightend = 0
-        init = score_suite_list[0].info_score
+        if (get_informed):
+            init = informed_threshold
+        else:
+            init = score_suite_list[0].info_score
 
         for sub in score_suite_list:
             sub_info = sub.info_score
-            if sub_info == init:
+            if sub_info <= init:
                 rightend += 1
             else:
                 break
-
-        lowest_info = score_suite_list[0:rightend]
-        random.shuffle(lowest_info)
         other_info = score_suite_list[rightend:]
-        info_submissions = lowest_info + other_info
+        if (get_informed):
+            info_submissions = other_info
+        else:
+            lowest_info = score_suite_list[0:rightend]
+            random.shuffle(lowest_info)
+            info_submissions = lowest_info + other_info
 
         # return list of keys of dictionary of song objects sorted by info
         return info_submissions
 
+    @staticmethod
+    def weighted_lowest_info(
+            score_suite_list: List["ScoreSuite"],
+            luck_factor : int) -> "ScoreSuite":
+        #this is a luck factor that determines how skewwed we are to low information songs
+        # when luck factor goes to 0 we will only grab the lowest infromed song
+        # when luck factor is 1 all songs have at some chance of being chosen while low informed songs are more likely 
+        # when luck factor 1>> all songs have roughly equal chance of being chosen
+        info_list = SongScores.get_info_sort(score_suite_list)
+        ceiling = round(((info_list[-1].info_score) + 1) * luck_factor)
+
+        random_num = 0
+        for song_score in info_list:
+            increase_random = ceiling - song_score.info_score
+            if (increase_random > 0):
+                random_num += increase_random
+            else:
+                break
+        select_num = random.randrange(random_num + 1)
+        ceiling_check = 0
+
+        for song_score in enumerate(info_list):
+            increase_ceiling = ceiling - info_list[song_score[0]].info_score
+            if (increase_ceiling < 0):
+                increase_ceiling = 0
+
+            if ((ceiling_check + increase_ceiling) >= select_num):
+                return song_score[1]
+            else:
+                ceiling_check += increase_ceiling
```

Comment:
```suggestion
        for song_score in info_list:
            increase_ceiling = ceiling - song_score.info_score
            if (increase_ceiling < 0):
                increase_ceiling = 0

            if ((ceiling_check + increase_ceiling) >= select_num):
                return song_score
            else:
                ceiling_check += increase_ceiling
```

Needless use of `enumerate`.

Reply:
yeah you're right this is a remnant of when I was using the index for some part of the weighting.